spp_import_match/models/import_match.py

```diff
@@ -1,126 +1,149 @@
+                            row_value = tuple_val[field.sub_field_id.name]
+                            add_to_domain = True
+                            field_value = field.field_id.name + "." + field.sub_field_id.name
+                    if add_to_domain:
+                        domain.append((field_value, "=", row_value))
+            if not combination_valid:
+                continue
+            match = model.search(domain)
+            if len(match) == 1:
+                return match
+
+        return model
+
+    @api.model
+    @tools.ormcache("model_name", "frozenset(fields)")
+    def _usable_rules(self, model_name, fields):
+        result = self
+        available = self.search([("model_name", "=", model_name)])
+        field_to_match = []
+        for record in available:
+            field_to_match.append(record.field_ids.mapped("name"))
+            for f in record.field_ids:
+                if f.name in fields or f.field_id.name in fields:
+                    result |= record
+
+        return result.ids, field_to_match
+
+
+class SPPImportMatchFields(models.Model):
+    _name = "spp.import.match.fields"
+    _description = "Fields for Import Matching"
+
+    name = fields.Char(compute="_compute_name")
+    field_id = fields.Many2one(
+        "ir.model.fields",
+        string="Field",
+        required=True,
+        ondelete="cascade",
+        domain="[('model_id', '=', model_id)]",
+        help="Fields to Match in Importing",
+    )
+    relation = fields.Char(related="field_id.relation")
+    sub_field_id = fields.Many2one(
+        "ir.model.fields",
+        string="Sub-Field",
+        ondelete="cascade",
+        help="Sub Fields to Match in Importing",
+    )
+    match_id = fields.Many2one("spp.import.match", string="Match", ondelete="cascade")
+    model_id = fields.Many2one(related="match_id.model_id")
+    conditional = fields.Boolean()
+    imported_value = fields.Char(
+        help="This will be used as a condition to disregard this field if matched"
+    )
+
+    def _compute_name(self):
+        for rec in self:
+            name = rec.field_id.name
+            if rec.sub_field_id:
+                name = rec.field_id.name + "/" + rec.sub_field_id.name
+            rec.name = name
+
+    @api.onchange("field_id")
+    def _onchange_field_id(self):
+        for rec in self:
+            field_id = rec.field_id.id
+            fields_list = []
+            for field in rec.match_id.field_ids:
+                new_id_str = str(field.id)
+                if not new_id_str.find("NewId_'virtual"):
+                    fields_list.append(field.field_id.id)
+
+            duplicate_counter = 0
+            for duplicate_field in fields_list:
+                if duplicate_field == field_id:
+                    duplicate_counter += 1
+
+            if duplicate_counter > 1:
+                raise ValidationError(_("Field '%s', already exists!") % rec.field_id.field_description)
+

```

Comment:
```suggestion
_sql_constraints = [
    (
        "field_id_uniq",
        "UNIQUE(field_id, match_id)",
        "Field should be unique for each match",
    )
]


```

Reply:
This is edited, sql_constraints is not good here is we should allow different sub-fields for a one2many field.